### PR TITLE
Apply latest suggested project settings

### PR DESCRIPTION
**Pull request by webframp**
As suggested by Xcode 11.4.1, using defaults

### CODE REVIEW

**Comment by webframp** [comment]
Just posting this before I go to sleep, so I remember to come back to it.

FYI @dive only two warnings currently, both in https://github.com/MobileOrg/mobileorg/blob/develop/Classes/Utilities/UIAlertController%2BMobileOrg.swift#L47

`Cast from 'UIViewController?.Type' to unrelated type 'UINavigationController.Type' always fails`

Will try to find time this week to resolve.

File: MobileOrg.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1086,6 +1112,32 @@
...
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
```
Comment (dive) [inline MobileOrg.xcodeproj/project.pbxproj]: The same as above.

File: MobileOrg.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1073,6 +1073,32 @@
...
```
Comment (dive) [inline MobileOrg.xcodeproj/project.pbxproj]: We already have all these warnings enabled in the `Global.shared.xcconfig`. It seems that Xcode just ignores the `.xconfig` when suggesting such changes. I think we can revert them, bump for the `LastUpgradeCheck` will resolve this warning from Xcode.

File: MobileOrg.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1099,6 +1151,34 @@
...
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
+				ONLY_ACTIVE_ARCH = YES;
```
Comment (dive) [inline MobileOrg.xcodeproj/project.pbxproj]: The same as above.

**Review by dive — COMMENTED** [review]
Thanks! 

Yeah, it seems that we have to rewrite the `mo_presentingViewController` to avoid these "Cast always fails" warnings from the compiler. I suggest something like this:

```swift
private static var mo_presentingViewController: UIViewController? {
    let rootViewController = UIApplication.shared.delegate?.window??.rootViewController
    if let controller = (rootViewController as? UINavigationController)?.topViewController {
        return controller
    } else if let controller = (rootViewController as? UITabBarController)?.selectedViewController {
        return controller
    }
    return rootViewController
}
```

File: MobileOrg.xcodeproj/project.pbxproj (hunk 2 above)
Comment (webframp) [reply MobileOrg.xcodeproj/project.pbxproj]: Makes sense to me. A bit annoying that it ignores the `.xconfig` in this case

File: MobileOrg.xcodeproj/project.pbxproj (hunk 3 above)
Comment (webframp) [reply MobileOrg.xcodeproj/project.pbxproj]: Removed

File: MobileOrg.xcodeproj/project.pbxproj (hunk 1 above)
Comment (webframp) [reply MobileOrg.xcodeproj/project.pbxproj]: Removed

File: MobileOrg.xcodeproj/project.pbxproj (hunk 2 above)
Comment (webframp) [reply MobileOrg.xcodeproj/project.pbxproj]: Good suggestion, bumped check version in ea63d7165bb0eee51802497f007bb7d71f30943e

Always hate dealing with project files "managed" by Xcode

**Comment by webframp** [comment]
@dive updated based on feedback, thanks. I think this is good to go if you can take one last look.

**Comment by webframp** [comment]
Tested this a bit more, seems fine and warnings are gone after last 2 commits here. Going to merge.

**Pull request merged**